### PR TITLE
Fix java.util.NoSuchElementException at com.mrousavy.camera.extensions.RecordingSession_getRecommendedBitRateKt.getRecommendedBitRate(RecordingSession+getRecommendedBitRate.kt:107)

### PR DESCRIPTION
**Pull request by Romick2005**
Android Xiaomi Redmi Note 11 Pro 5G issue:
```
Short message: An unknown error occurred while trying to start a video recording! null.
 ERROR  {"cause": {"message": null, "stacktrace": "java.util.NoSuchElementException
	at com.mrousavy.camera.extensions.RecordingSession_getRecommendedBitRateKt.getRecommendedBitRate(RecordingSession+getRecommendedBitRate.kt:107)
	at com.mrousavy.camera.core.RecordingSession.getBitRate(RecordingSession.kt:132)
	at com.mrousavy.camera.core.RecordingSession.<init>(RecordingSession.kt:37)
	at com.mrousavy.camera.core.CameraSession.startRecording(CameraSession.kt:532)
	at com.mrousavy.camera.CameraView_RecordVideoKt.startRecording(CameraView+RecordVideo.kt:34)
	at com.mrousavy.camera.CameraViewModule$startRecording$1.invokeSuspend(CameraViewModule.kt:91)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
"}, "code": "capture/unknown", "message": "An unknown error occurred while trying to start a video recording! null", "userInfo": null}
```

### CODE REVIEW

**Comment by Romick2005** [comment]
Related with [Fix VideoProfile possibly being null #2231](https://github.com/mrousavy/react-native-vision-camera/pull/2231#issuecomment-1843465890)

File: package/android/src/main/java/com/mrousavy/camera/CameraDevicesManager.kt [hunk 1]
```diff
@@ -71,8 +71,6 @@ class CameraDevicesManager(private val reactContext: ReactApplicationContext) :
     eventEmitter.emit("CameraDevicesChanged", getDevicesJson())
   }
 
-  override fun hasConstants(): Boolean = true
```
Comment (mrousavy) [inline package/android/src/main/java/com/mrousavy/camera/CameraDevicesManager.kt]: Please revert this change

File: package/android/src/main/java/com/mrousavy/camera/extensions/RecordingSession+getRecommendedBitRate.kt [hunk 2]
```diff
@@ -31,7 +31,7 @@ fun RecordingSession.getRecommendedBitRate(fps: Int, codec: VideoCodec, hdr: Boo
   if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
     val profiles = CamcorderProfile.getAll(cameraId, quality)
     if (profiles != null) {
-      val best = profiles.videoProfiles.filterNotNull().minBy {
+      val best = profiles.videoProfiles.filterNotNull().minByOrNull {
```
Comment (mrousavy) [inline package/android/src/main/java/com/mrousavy/camera/extensions/RecordingSession+getRecommendedBitRate.kt:34]: Good stuff, makes sense!

**Review by mrousavy — CHANGES_REQUESTED** [review]
Hey - thanks for your PR! Please revert the unrelated change for now, there's a different PR for that.

The other change looks good, thanks!

File: package/android/src/main/java/com/mrousavy/camera/CameraDevicesManager.kt (hunk 1 above)
Comment (Romick2005) [reply package/android/src/main/java/com/mrousavy/camera/CameraDevicesManager.kt]: Done, sir.

**Comment by mrousavy** [comment]
Awesome, thanks man - LGTM!

**Pull request merged**